components/director/internal/tenantmapping/model.go

```diff
@@ -0,0 +1,78 @@
+package tenantmapping
+
+import (
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/pkg/errors"
+)
+
+// CredentialSrc wraps possible sources of credentials like token, username and certificate
+type CredentialSrc string
+
+// CredenialRefObjectType wraps possible options for the origin object represented by the credential
+type CredenialRefObjectType string
+
+func sysAuthToCredRefObjectType(sysAuth model.SystemAuth) CredenialRefObjectType {
+	if sysAuth.AppID != nil {
+		return ApplicationRefObject
+	} else if sysAuth.RuntimeID != nil {
+		return RuntimeRefObject
+	} else if sysAuth.IntegrationSystemID != nil {
+		return IntegrationSystemRefObject
+	}
+
+	return ""
+}
+
+const (
+	ApplicationRefObject       CredenialRefObjectType = "Application"
+	RuntimeRefObject           CredenialRefObjectType = "Runtime"
+	IntegrationSystemRefObject CredenialRefObjectType = "Integration System"
+	UserRefObject              CredenialRefObjectType = "User"
+
+	CertificateCredSrc CredentialSrc = "Certificate"
+	TokenCredSrc       CredentialSrc = "Token"
```

Comment:
In current implementation it describes just the OAuth 2.0 token case, so I would rename it